blog/users/views.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from users.serializers import UserSignUpSerializer
+
+
+@api_view(['POST',])
+def registration_view(request):
+    if request.method == 'POST':
+        serializer = UserSignUpSerializer(data=request.data)
+        data = {}
+        if serializer.is_valid():
+            user = serializer.save()
+            data['response'] = 'successfully registered a new user'
+            data['username'] = user.username
+        else:
```

Comment:
you dont need this` if else`, instead use `serializer.is_valid(raise_exception=True)`